api/types/role.go

```diff
@@ -680,50 +680,6 @@ func (r *Rule) CheckAndSetDefaults() error {
 	return nil
 }
 
-// score is a sorting score of the rule, the larger the score, the more
-// specific the rule is
-func (r *Rule) score() int {
-	score := 0
-	// wildcard rules are less specific
-	if utils.SliceContainsStr(r.Resources, Wildcard) {
-		score -= 4
-	} else if len(r.Resources) == 1 {
-		// rules that match specific resource are more specific than
-		// fields that match several resources
-		score += 2
-	}
-	// rules that have wildcard verbs are less specific
-	if utils.SliceContainsStr(r.Verbs, Wildcard) {
-		score -= 2
-	}
-	// rules that supply 'where' or 'actions' are more specific
-	// having 'where' or 'actions' is more important than
-	// whether the rules are wildcard or not, so here we have +8 vs
-	// -4 and -2 score penalty for wildcards in resources and verbs
-	if len(r.Where) > 0 {
-		score += 8
-	}
-	// rules featuring actions are more specific
-	if len(r.Actions) > 0 {
-		score += 8
-	}
-	return score
-}
-
-// IsMoreSpecificThan returns true if the rule is more specific than the other.
-//
-// * nRule matching wildcard resource is less specific
-// than same rule matching specific resource.
-// * Rule that has wildcard verbs is less specific
-// than the same rules matching specific verb.
-// * Rule that has where section is more specific
-// than the same rule without where section.
-// * Rule that has actions list is more specific than
-// rule without actions list.
-func (r *Rule) IsMoreSpecificThan(o Rule) bool {
```

Comment:
Curious why did this (along with `score()` above) move back to lib/services?

Reply:
This function (and others) is business logic that shouldn't be in `api/types`. Since these PRs are trying to reduce dependencies and reduce the overall size of `api`, I moved this back. I should have made this a separate PR though.